code/src/store/api/productsApi.ts

```diff
@@ -6,10 +6,27 @@ import { URLS } from "@/constants/requests";
 
 import { Product } from "@/types/product.types";
 
+type GetProductsResponse = {
+  items: Product[];
+  pagesCount: number;
+  itemsCount: number;
+};
+
+type GetProductsParams = {
+  page?: number;
+  itemsPerPage?: number;
+};
+
 const productsApi = appApi.injectEndpoints({
   endpoints: (build) => ({
-    getProducts: build.query<Product[], void>({
-      query: () => URLS.products.get
+    getProducts: build.query<
+      GetProductsResponse,
+      GetProductsParams | undefined | void
+    >({
```

Comment:
Why undefined | void?

Reply:
I used it to overcome ts error. I removed undefined and fixed ts error in other, more convenient way